packages/lib/src/vis/shared/SelectionTool.tsx

```diff
@@ -27,15 +28,15 @@ function SelectionTool(props: Props) {
 
   const [startPoint, setStartPoint] = useState<Vector2>();
   const [endPoint, setEndPoint] = useState<Vector2>();
+  const [modifierKeyPressed, setModifierPressed] = useState(!modifierKey);
```

Comment:
`modifierKeyPressed` is initialized to `true` if no `modifierKey` is given. Otherwise, we can't select at first render.  

When `modifierKey` is given, I initialize to `false` so that it can be set to `true` when the appropriate `keyDown` event is fired.

Reply:
I would maybe rename this state `isAllowed` - i.e. `const [isAllowed, toggleAllowed] = useToggle(!modifierKey)`, or `isSelectionAllowed` if it's too vague, to better cover the cases where no modifier key is passed and where the wrong modifier key is pressed.

Reply:
I renamed to `isAllowed` but wasn't too sure if the implementation using `toggle` was going to work. I did it in a separate commit and it appears to work well.

Tell me what you prefer